docs/src/pages/components/date-picker/SubComponentsPickers.tsx

```diff
@@ -1,18 +1,56 @@
 import * as React from 'react';
 import AdapterDateFns from '@material-ui/lab/AdapterDateFns';
-import LocalizaitonProvider from '@material-ui/lab/LocalizationProvider';
+import LocalizationProvider from '@material-ui/lab/LocalizationProvider';
+import Box from '@material-ui/core/Box';
+import { makeStyles } from '@material-ui/core/styles';
 import CalendarPicker from '@material-ui/lab/CalendarPicker';
+import MonthPicker from '@material-ui/lab/MonthPicker';
+import YearPicker from '@material-ui/lab/YearPicker';
+
+const minDate = new Date('2000-01-01T00:00:00.000');
+const maxDate = new Date('2100-01-01T00:00:00.000');
+
+const useYearPickerStyles = makeStyles({
+  root: {
+    maxWidth: '30ch',
+  },
+});
 
 export default function SubComponentsPickers() {
```

Comment:
The layout seems broken on this demo. For instance, shouldn't we have an overflow here:

<img width="367" alt="Screenshot 2021-04-18 at 16 40 58" src="https://user-images.githubusercontent.com/3165635/115149533-dffcea00-a064-11eb-9f66-af65d71833e6.png">

Reply:
I don't understand what you mean. Please elaborate why this layout seems broken first. Then we can talk about what should overflow into what.

Reply:
1. 1800 pixels tall demo. It's impossible to have the source "side by side" with this content. A max-height on the year picker would be a great first step


![scroll](https://user-images.githubusercontent.com/3165635/115314409-6f91be00-a175-11eb-8bac-60a7a3bcebf7.gif)

2. It overflows on smaller screens
<img width="771" alt="Screenshot 2021-04-20 at 01 16 06" src="https://user-images.githubusercontent.com/3165635/115314699-02325d00-a176-11eb-8069-ed90439457ae.png">

3. The CalendarPicker flexbox shrinks. It shouldn't. I think that we miss a flexShrink: 0 in its source. It looks significantly distorted:

<img width="284" alt="Screenshot 2021-04-20 at 01 18 21" src="https://user-images.githubusercontent.com/3165635/115314886-63f2c700-a176-11eb-88b1-57d9f946feff.png">

The same goes for the 2 other components.

Reply:
Limited to just a couple of years.